.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+echo "Running pre-commit tasks..." >&2
+npm run lint:check || (echo 'Lint failed. Fix the above errors, and/or run `npm  run lint` to autofix where possible, then re-commit.' >&2 && exit 1)
+npm run format:check || (echo 'Bad code formatting. Run `npm run format`, then re-commit.' >&2 && exit 1)
+npm run test-compile || (echo 'Test compilation failed. Fix the above errors, then recommit.' >&2 && exit 1)
```

Comment:
verbalism

```suggestion
npm run lint:check
npm run format:check
npm run test-compile
```

Reply:
I put these in because it's a little more friendly to new contributors. The output of `npm run lint` is going to suggest using `--fix`, which won't really be useful, since `--check` and `--fix` can't coexist AFAIK.

Ditto with Prettier. It isn't obvious what command you need to run to fix it.

Reply:
adding hints is useful. but `'Test compilation failed. Fix the above errors, then recommit.'` isn't helping anyone, it's noise

Reply:
I dunno, I won't push too hard on this, and if you feel strongly I'll cut down on the wording, but I do think it's nice for those unfamiliar with the commands whizzing by to know what stage failed (pre-commit did this with fancy terminal art, but that was a bit much for this). 

Difference of opinion :)

Reply:
what contributors are you envisioning that will be committing code contributions but are unfamiliar with `npm run test-compile` ?

Reply:
None, it's just a lot of commands whizzing by on commit, and can make their lives a bit easier. 

It seems you do feel strongly, so I'll tone it down a bit